kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala

```diff
@@ -101,6 +101,13 @@ case class KyuubiConf(loadSysDefault: Boolean = true) extends Logging {
     settings.entrySet().asScala.map(x => (x.getKey, x.getValue)).toMap[String, String]
   }
 
+  /** Get all engine envs as map */
+  def getEngineEnvs: Map[String, String] = {
+    settings.entrySet().asScala.filter(_.getKey.startsWith(KYUUBI_ENGINE_ENV_PREFIX))
+      .map(entry => entry.getKey.stripPrefix(KYUUBI_ENGINE_ENV_PREFIX) -> entry.getValue)
```

Comment:
use getAllWithPrefix?

Reply:
this should also lookup `sys.env` first and apply `KYUUBI_ENGINE_ENV_PREFIX`s in the rightmost way